CRM/Core/Smarty/plugins/prefilter.htxtFilter.php

```diff
@@ -22,6 +22,10 @@ function smarty_prefilter_htxtFilter($tpl_source, &$smarty) {
       $htxts++;
       return sprintf('{if $id == %s}%s', $m[1], $m[0]);
     },
+    '/\{htxt id=(\$\w+)}/' => function ($m) use (&$htxts) {
```

Comment:
@larssandergreen That's correct, it doesn't/didn't match ` otherstuff`. 

Regarding this specific example, we could treat either `}` or ` ` as terminating the `id=...` phrase. This would allow subsequent parameters.

I wouldn't exactly drop the `}`. Consider text like `{htxt id=$foo.bar}` or `{htxt id=$foo[bar]}` or `{htxt id=$foo"bar"}` or `{htxt id=$foo{bar}}`:
* If the regex has no closing marker (`id=(\$\w+)`), then it does a partial match -- matching `id=$foo` and ignoring `.bar` or `[bar]` or `"bar"` or `{bar}`. I expect that to produce some weird bug.
* If the regex has a closing marker (`id=(\$\w+)[} ]`), then it will not match any of those examples. `htxtFilter` will throw an exception (`Invalid {htxt} tag`), which points you to the right element.

Within the world of "unsupported edge-cases", the explicit error is better, no?

Anyway, the example you gave (` otherstuff`) should be fairly easy to accommodate by accepting ` ` as another closing-marker. I'm pushing up a commit to do that.

Reply:
Makes sense.